keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusServiceTests.kt

```diff
@@ -30,238 +29,213 @@ import com.netflix.spinnaker.keel.persistence.ResourceStatus.MISSING_DEPENDENCY
 import com.netflix.spinnaker.keel.persistence.ResourceStatus.RESUMED
 import com.netflix.spinnaker.keel.persistence.ResourceStatus.UNHAPPY
-import com.netflix.spinnaker.keel.test.combinedInMemoryRepository
-import com.netflix.spinnaker.keel.test.deliveryConfig
 import com.netflix.spinnaker.keel.test.resource
 import com.netflix.spinnaker.kork.exceptions.SpinnakerException
-import com.netflix.spinnaker.time.MutableClock
 import dev.minutest.junit.JUnit5Minutests
 import dev.minutest.rootContext
 import io.mockk.every
-import io.mockk.slot
+import io.mockk.mockk
 import strikt.api.expectThat
 import strikt.assertions.isEqualTo
 
 class ResourceStatusServiceTests : JUnit5Minutests {
   companion object Fixture {
-    val clock = MutableClock()
-    val repository = combinedInMemoryRepository(clock)
-    val actuationPauser = ActuationPauser(repository.resourceRepository, repository.pausedRepository, repository.publisher, clock)
-    val subject = ResourceStatusService(repository.resourceRepository, actuationPauser)
+    val repository = mockk<ResourceRepository>()
+    val actuationPauser = mockk<ActuationPauser>()
+    val subject = ResourceStatusService(repository, actuationPauser)
     val resource = resource()
-    val deliveryConfig = deliveryConfig(resource)
     val user = "keel@keel.io"
+
+    val events = mutableListOf<ResourceHistoryEvent>()
   }
 
   fun tests() = rootContext<Fixture> {
     fixture {
       Fixture
     }
 
-    before {
-      val event = slot<PersistentEvent>()
-      every {
-        repository.publisher.publishEvent(capture(event))
-      } answers {
-        if (event.captured is ApplicationEvent) {
-          repository.appendApplicationHistory(event.captured as ApplicationEvent)
-        } else {
-          repository.appendResourceHistory(event.captured as ResourceEvent)
-        }
-      }
-    }
-
-    after {
-      repository.dropAll()
-      clock.reset()
-    }
-
     context("a delivery config with a resource exists") {
       before {
-        repository.upsertDeliveryConfig(deliveryConfig)
+        every { repository.get(resource.id) } returns resource
+        every { repository.eventHistory(resource.id) } answers { events.reversed() }
       }
 
       context("resource status") {
         context("when resource is paused") {
           before {
-            actuationPauser.pauseResource(resource.id, user)
+            every { actuationPauser.isPaused(resource.id) } returns true
           }
 
           test("returns PAUSED") {
             expectThat(subject.getStatus(resource.id)).isEqualTo(PAUSED)
           }
         }
 
-        context("when resource's parent application is paused") {
+        context("when resource is not paused") {
           before {
-            actuationPauser.pauseApplication(resource.application, user)
-          }
-
-          test("returns PAUSED") {
-            expectThat(subject.getStatus(resource.id)).isEqualTo(PAUSED)
-          }
-        }
```

Comment:
Now I'm stubbing `ActuationPauser` there's no difference between an application-level pause and a resource-level pause in the context of this test.